common/oatbox/service/ServiceManager.php

```diff
@@ -63,25 +68,69 @@ public function __construct($configService)
      *
      * @param string $serviceKey
      * @return ConfigurableService
+     * @throws ServiceNotFoundException
+     * @see ContainerInterface::get()
      */
     public function get($serviceKey)
     {
-        if ((interface_exists($serviceKey) || class_exists($serviceKey)) && defined($serviceKey . '::SERVICE_ID')) {
-            $serviceKey = $serviceKey::SERVICE_ID;
+        $serviceId = $this->getServiceId($serviceKey);
+        if (!isset($this->services[$serviceId])) {
+            $this->services[$serviceId] = $this->load($serviceId, $serviceKey);
+        }
+        return $this->services[$serviceId];
+    }
+
+    /**
+     * Extract the service id from the provided parameter
+     * @param string $serviceKey
+     * @return string
+     */
+    protected function getServiceId($serviceKey) {
+        return ((interface_exists($serviceKey) || class_exists($serviceKey)) && defined($serviceKey . '::SERVICE_ID'))
+            ? $serviceKey::SERVICE_ID
+            : (string)$serviceKey
+        ;
+    }
+
+    /**
+     * Loads the service referenced by id
+     *
+     * @param string $serviceId
+     * @param string $serviceKey
+     * @throws ServiceNotFoundException
+     * @return ConfigurableService
+     */
+    protected function load($serviceId, $serviceKey)
+    {
+        $service = $this->getConfig()->get($serviceId);
+        if ($service === false) {
+            $service = $this->tryAutowiring($serviceId, $serviceKey);
         }
-        if (!isset($this->services[$serviceKey])) {
-            $service = $this->getConfig()->get($serviceKey);
-            if ($service === false) {
-                throw new ServiceNotFoundException($serviceKey);
-            }
-            $this->services[$serviceKey] = $this->propagate($service);
+        if(is_object($service) &&  ($service instanceof ServiceLocatorAwareInterface)){
```

Comment:
There is no need to check if `$service` is an object. Just checking `$service instanceof ServiceLocatorAwareInterface` will give the desired result.

Reply:
Also to follow the PSR-2 standard the if keyword should have one space after it, and one space before the opening bracket. `if ($service instanceof ServiceLocatorAwareInterface) {`

Reply:
is_object check was added to ensure no  warnings are thrown here, and though it might not be required anymore I don't feel confident enough to remove it